src/test_typing_extensions.py

```diff
@@ -1645,6 +1645,37 @@ def test_final_forward_ref(self):
         self.assertNotEqual(gth(Loop, globals())['attr'], Final[int])
         self.assertNotEqual(gth(Loop, globals())['attr'], Final)
 
+    def test_annotation_and_optional_default(self):
```

Comment:
Do you think that more tests are necessary? Do you have ideas where something could go wrong?